src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -98,12 +100,138 @@ static void AppendShardSizeMinMaxQuery(StringInfo selectQuery, uint64 shardId,
 static void AppendShardSizeQuery(StringInfo selectQuery, ShardInterval *shardInterval,
 								 char *quotedShardName);
 
+static HeapTuple CreateDiskSpaceTuple(TupleDesc tupleDesc, uint64 availableBytes,
+									  uint64 totalBytes);
+static bool GetLocalDiskSpaceStats(uint64 *availableBytes, uint64 *totalBytes);
+
 /* exports for SQL callable functions */
+PG_FUNCTION_INFO_V1(citus_local_disk_space_stats);
 PG_FUNCTION_INFO_V1(citus_table_size);
 PG_FUNCTION_INFO_V1(citus_total_relation_size);
 PG_FUNCTION_INFO_V1(citus_relation_size);
 PG_FUNCTION_INFO_V1(citus_shard_sizes);
 
+
+/*
+ * CreateDiskSpaceTuple creates a tuple that is used as the return value
+ * for citus_local_disk_space_stats.
+ */
+static HeapTuple
+CreateDiskSpaceTuple(TupleDesc tupleDescriptor, uint64 availableBytes, uint64 totalBytes)
+{
+	Datum values[TABLE_METADATA_FIELDS];
+	bool isNulls[TABLE_METADATA_FIELDS];
+
+	/* form heap tuple for remote disk space statistics */
+	memset(values, 0, sizeof(values));
+	memset(isNulls, false, sizeof(isNulls));
+
+	values[0] = UInt64GetDatum(availableBytes);
+	values[1] = UInt64GetDatum(totalBytes);
+
+	HeapTuple diskSpaceTuple = heap_form_tuple(tupleDescriptor, values, isNulls);
+
+	return diskSpaceTuple;
+}
+
+
+/*
+ * citus_local_disk_space_stats returns total disk space and available disk
+ * space for the disk that contains PGDATA.
+ */
+Datum
+citus_local_disk_space_stats(PG_FUNCTION_ARGS)
+{
+	uint64 availableBytes = 0;
+	uint64 totalBytes = 0;
+
+	if (GetLocalDiskSpaceStats(&availableBytes, &totalBytes))
+	{
+		ereport(WARNING, (errmsg("could not get disk space")));
```

Comment:
On my mac, I get the following output. Do you have any clue on this?
```

[local] onderkalaci@postgres:5432-24984=# select * from  citus_local_disk_space_stats();
WARNING:  could not get disk space
 available_disk_size | total_disk_size 
---------------------+-----------------
        171613466624 |    499963174912
(1 row)

Time: 0.530 ms
[local] onderkalaci@postgres
```

Reply:
I guess we should have if (!GetLocalDiskSpaceStats()) warning

Reply:
Do you get the same warning in https://github.com/citusdata/citus/pull/4383?

Reply:
I think you will get the same warning there because this check should be:

```
if (!GetLocalDiskSpaceStats(&availableBytes, &totalBytes))
```

Not

```
if (GetLocalDiskSpaceStats(&availableBytes, &totalBytes))
```

Reply:
I think it was just a missing `!`, I don't think it was your local machine :)